### PR TITLE
feature(config): add HAR file config definition 

### LINKED ISSUE #274: HAR support
Monika should be able to read [HAR (HTTP Archive) format](https://en.wikipedia.org/wiki/HAR_(file_format)) and automatically probe using the requests defined in the HAR file.

- [x] Add a new flag: `monika --har <path-to-har-data>`
- [ ] The code or function to convert postman data to monika configuration should be exportable. 
- [x] Add tests to make sure conversion runs as expected.
- [ ] Add docs.

### PR DESCRIPTION
**Pull request by ilmiawan**
# Monika PR
Fix #274 
## What issue this PR is trying to solve
Currently the configuration is only based on monika.json specific config. Need to add capability to read configuration from HAR file.
## How this PR solve the issue
Add capability to run probe read from HAR file definition.
## How to test
1. run monika with `--har` flag set with the path to the HAR file 
2. It should be running well to monitor url set from HAR file

### CODE REVIEW

File: docs/src/pages/guides/probes.md [hunk 1]
```diff
@@ -112,6 +112,10 @@ In a configuration with multiple probes, `Monika` will perform the requests in s
 
 On completion, `Monika` will sleep until the next interval to start again. At the top of the `monika.json` file there is an `interval` setting. The execution will be restarted after every `interval`. If interval is shorter than the amount of time to dispatch all the requests, then `Monika` will immediately repeat after the last probe response and any notification alerts sent. When the `--repeat` flag is set with a value, Monika will not run indefinitely, instead, it will stop after executing the probes as many times as specified.
 
+## HAR file support
+
+HAR [HTTP-Archive](<https://en.wikipedia.org/wiki/HAR_(file_format)>) format is created by Web Performance Working Group and become the standard use in browser archive request data definition. Currently `Monika` are able to run using HAR format, `--har` flag is set with path to the har definition file.
```
Comment (sapiderman) [inline docs/src/pages/guides/probes.md:117]: format was created by the Web Performance Working Group and has become the standard  in browser archive request data definition. Currently `Monika` is able to use this HAR format, use the  `--har` flag with path to the HAR definition file.

**Review by sapiderman — CHANGES_REQUESTED** [review]

File: docs/src/pages/guides/probes.md (hunk 1 above)
Comment (nicnocquee) [inline docs/src/pages/guides/probes.md:117]: Replace this sentence:

Currently `Monika` are able to run using HAR format, `--har` flag is set with path to the har definition file.

with

To run monika using a HAR file, use `--har` flag as follows:

```
monika --har <path_to_HAR_file>
```

File: src/components/config/index.ts [hunk 2]
```diff
@@ -82,34 +83,65 @@ const handshakeAndValidate = async (config: Config) => {
...
-export const setupConfigFromUrl = async (
-  url: string,
-  checkingInterval: number
-) => {
-  const fetched = await fetchConfig(url)
-  await handshakeAndValidate(fetched)
-  cfg = fetched
+const parseHarAndValidate = async (path: string) => {
+  const parsed = parseHarFile(path)
+  await handshakeAndValidate(parsed)
```
Comment (nicnocquee) [inline src/components/config/index.ts:131]: I don't think we need to do "handshake" for the HAR file. Monika performs handshake with Symon if the config is specified in monika.json. There won't be such configuration in a HAR file.

File: src/components/config/parse.ts [hunk 3]
```diff
@@ -47,3 +49,47 @@ export const parseConfig = (configPath: string): Config => {
...
+
+    const harConfig: Config = {
+      probes: [harProbe],
+    }
+
+    return harConfig
+  } catch (error) {
+    if (error.code === 'ENOENT' && error.path === filepath) {
+      throw new Error(
+        'Har file not found. By default, Monika looks for monika.json configuration file in the current directory.\n\nOtherwise, you can also specify a configuration file using -c flag as follows:\n\nmonika -c <path_to_configuration_file>\n\nYou can create a configuration file via web interface by opening this web app: https://hyperjumptech.github.io/monika-config-generator/'
```
Comment (nicnocquee) [inline src/components/config/parse.ts:85]: The error can be just "HAR file not found." The rest is not needed in this case.

**Review by nicnocquee — CHANGES_REQUESTED** [review]

File: docs/src/pages/guides/probes.md (hunk 1 above)
Comment (ilmiawan) [reply docs/src/pages/guides/probes.md]: fixed

File: src/index.ts [hunk 4]
```diff
@@ -137,6 +136,12 @@ class Monika extends Command {
       description: 'specific probe ids to run',
       multiple: false,
     }),
+
+    har: flags.string({
+      char: 'H', // (H)ar file to
+      description: 'add Har file to read',
```
[the remark this answers is not in the document]
Comment (ilmiawan) [reply src/index.ts]: Fixed

File: docs/src/pages/guides/probes.md (hunk 1 above)
Comment (ilmiawan) [reply docs/src/pages/guides/probes.md]: fixed

File: src/components/config/parse.ts (hunk 3 above)
Comment (ilmiawan) [reply src/components/config/parse.ts]: Fixed

File: src/components/config/index.ts (hunk 2 above)
Comment (ilmiawan) [reply src/components/config/index.ts]: removed handhshake

**Review by nicnocquee — APPROVED** [review]

File: src/components/config/index.ts [hunk 5]
```diff
@@ -82,34 +83,58 @@ const handshakeAndValidate = async (config: Config) => {
   }
 }
 
-export const setupConfigFromFile = async (path: string, watch: boolean) => {
```
Comment (raosan) [inline src/components/config/index.ts:85]: need to revert and simplify this function to be in sync with postman config https://github.com/hyperjumptech/monika/pull/286

**Review by raosan — CHANGES_REQUESTED** [review]

**Review by sapiderman — APPROVED** [review]

File: src/components/config/index.ts (hunk 5 above)
Comment (ilmiawan) [reply src/components/config/index.ts:85]: refactored

File: src/components/config/parse.ts [hunk 6]
```diff
@@ -24,13 +24,56 @@
 
 import { Config } from '../../interfaces/config'
 import { readFileSync } from 'fs'
+import { RequestConfig } from '../../interfaces/request'
+import { Probe } from '../../interfaces/probe'
 
-export const parseConfig = (configPath: string): Config => {
+export const parseHarFile = (fileContents: string): Config => {
```
Comment (raosan) [inline src/components/config/parse.ts:30]: as mentioned by @nicnocquee here https://github.com/hyperjumptech/monika/pull/286#discussion_r673710410

it should be moved to their own file

File: src/components/config/index.ts [hunk 7]
```diff
@@ -105,8 +113,6 @@ export const setupConfigFromUrl = async (
   const fetched = await fetchConfig(url)
   await handshakeAndValidate(fetched)
   cfg = fetched
-  cfg.version = cfg.version || md5Hash(cfg)
```
[the remark this answers is not in the document]
Comment (ilmiawan) [reply src/components/config/index.ts]: forget to put it back. fixed.

File: src/components/config/parse.ts (hunk 6 above)
Comment (ilmiawan) [reply src/components/config/parse.ts]: done

**Review by raosan — APPROVED** [review]
lgtm

**Pull request merged**